CodeStarterCamp_Week3/Type.swift

```diff
@@ -7,42 +7,87 @@
 
 import Foundation
 
-struct Person {
+class Person {
     var name: String
     var money: UInt
     
-    func buySomething() {
-        print("\(self.name) buy something!")
+    init(name: String, money: UInt) {
+        self.name = name
+        self.money = money
+    }
+    
+    func buyCoffee(at coffeeShop: CoffeeShop, coffee: Coffee) {
+        guard (self.money > coffee.price) else {
+            print("잔액이 \(coffee.price - self.money)원만큼 부족합니다.")
+            return
+        }
+        
+        do {
+            try coffeeShop.order(coffee: coffee, nameOfCustomer: self.name)
+        } catch CoffeeShopError.noBaristaError { return }
+        catch {print("default ERROR")}
+    
+        self.money -= coffee.price
     }
 }
 
+
+class Barista: Person {
+    func makeCoffee(coffee: Coffee) {
+        print("\(coffee)를 만들었습니다!")
+    }
+}
+
+
 class CoffeeShop {
-    var sales: UInt
-    var barista: Person?
-    var menu: [Coffee]
-    let pickUpTable: String = "This is pick up table!"
+    private var sales: UInt = 0
+    private var menu: [Coffee] = []
+    var barista: Barista?
+    var pickUpTable: Coffee?
     
-    init() {
-        self.sales = 0
-        self.menu = [.americano(price: 4500), .cafeLatte(price: 5000)]
+    init(menu: [Coffee]) {
+        self.menu = menu
     }
     
-    func takeOrder(item: Coffee?) {
-        if case .some(let orderedItem) = item {
-            print(orderedItem)
-        } else { print("주문을 받지 않았습니다.") }
+    func isBaristaExist(barista: Barista?) -> Bool {
+        guard let _ = barista else {
+            return false
+        }
+        return true
     }
```

Comment:
```suggestion
    func isBaristaExist(barista: Barista?) -> Bool {
        guard barista != nil {
            return false
        }
        return true
    }
```

swift lint에서 권장하는 형식입니다!

Reply:
앗 이 코멘트를 늦게 확인했네요!
감사합니다! 

코멘트 내용대로 아까 연산프로퍼티로 바꿨던 부분을 아래와 같이 수정해도 되겠네요!

```Swift
var isBaristaExist: Bool{
        get {
            guard barista != nil else {
                return false
            }
            return true
        }
    }
```